Lib/ufo2ft/filters/flattenComponents.py

```diff
@@ -37,7 +37,7 @@ def filter(self, glyph):
 def _flattenComponent(component):
     """Returns list of tuples (baseGlyph, transform) of nested component."""
 
-    glyph = component.font[component.baseGlyph]
+    glyph = self.context.font[component.baseGlyph]
```

Comment:
where is `self` defined?

Reply:
you should add a `glyphSet` argument to that `_flattenComponents` function and pass in the `self.context.glyphSet`, not the font. The glyphSet will contain the (parent-less copies of the original glyphs that are being "filtered")

Reply:
Oops, the flattenComponents wasn’t meant to be here.

Reply:
ok but then we need to fix it in another PR

Reply:
you can't call `component.font` because the glyphs the filters operate on by default are copies without a parent (see preProcessor module where we make a copy of the font's glyph set and pass that as second argument to the filter functions; in your tests you're only testing the filter with one argument, which will modify a font in-place).